src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdater.scala

```diff
@@ -20,7 +20,7 @@ object InstanceUpdater extends StrictLogging {
 
     // We need to suspend reservation on already launched reserved instances
     // to prevent reservations being destroyed/unreserved.
-    val updatedReservation = if (updatedTask.status.condition == Condition.Reserved && !instance.reservation.exists(r => r.state.isInstanceOf[Reservation.State.Suspended])) {
+    val updatedReservation = if (updatedTask.status.condition.isTerminal && instance.hasReservation && !instance.reservation.exists(r => r.state.isInstanceOf[Reservation.State.Suspended])) {
```

Comment:
Hm, how about `instance.isReserved`/`updatedTask.status.isReserved` with

```scala
def isReserved = isTerminal && hasReservation
```

It might not apply in this case but in others.

Reply:
hmm I don't like the name `Reserved` to be honest, because in general sense reserved does not mean terminal. Instance that wasn't yet launched is also reserved in a sense...

But I would be totally open to other suggestions

Reply:
Ok, let's leave it for now.